apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,598 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+import {CodeBlock, Admonition} from 'ui'
+
+export const meta = {
+  id: 'auth-postgres-hooks',
+  title: 'Auth Hooks (Postgres)',
+  description: 'Use Supabase Postgres Functions to customize your authentication flow',
+}
+
+
+Supabase allows you to use PostgreSQL functions to alter the default Supabase Auth flow. Developers can use hooks to write custom behaviour when there are no built in ways of achieving the desired behaviour.
+
+Hooks might help you:
+
+- Track the origin of user signups by adding metadata
+- Improve security by adding additional checks to password and multi-factor authentication
+- Support legacy systems by integrating with identity credentials from external authentication systems
+- Add additional custom claims to your JWT
+
+
+You can use hooks at specific points along an Auth flow to perform custom behaviour via PostgreSQL Functions.  Currently, the following hooks are available:
+
+**Teams and** **Enterprise**
+
+- MFA Verification Attempt Hook
+- Password Verification Attempt Hook
+
+**Free**
+
+- Custom Access Token Hook
+
+
+Below is a flow chart illustrating how Supabase Auth makes use of an Auth Hook:
+
+<Mermaid
+ chart={`flowchart TD
+    subgraph Supabase Auth
+    Start[Event Occurs] --> DecisionA{Is a Hook defined?}
+
+    ValidityCheck --Yes-->ContinueFlow[Continue with flow]
+    ValidityCheck --No-->Error(Return Error)
+
+    DecisionA -- No ---->H[Continue with Supabase Default Behaviour]
+    H--> ValidityCheck{Is response valid?}
+    DecisionA --Yes--->ValidateInput((Validate Inputs))
+        ValidateOutput -->ContinueFlow
+    end
+
+
+
+    subgraph Auth Hook
+    ValidateInput ---->CallHook[Execute PostgreSQL Function]
+
+    CallHook ---> ValidateOutput((Validate Outputs))
+    end`}
+    />
+
+
+
+## Use Cases
```

Comment:
What I would suggest here is to:

- Split the "spec" from the "example" -- have one section explaining each hook's description, inputs and outputs (maybe a table would work? try it)
- In the examples section use tabs so the page does not get huge / busy.

Reply:
Further comments on this I think a structure similar to this one conveys the message better:

1. Page title and intro text
2. What are hooks? (Technical explanation, table of available hooks)
3. Creating and managing hooks
4. Available hooks
    1. Custom access token
        1. Inputs, outputs, best practices
        2. Examples (Tabs)
    3. MFA verification attempts
        1. Inputs, outputs, best practices
        2. Examples (Tabs)
    4. Password verification attempts
        1. Inputs, outputs, best practices
        2. Examples (Tabs)
5. Frequently asked questions